### PR TITLE
Disable mixed blocks in extensible variants

### PR DESCRIPTION
**Pull request by ncik-roberts**
Disallow mixed blocks in extensible variants (declared with either `+=` or `exception` syntaxes), by request of @lthls. This unblocks #2533.

This is mostly a few-line change in `typedecl.ml`, with a number of changes to tests. Most of the test updates are automatic changes I blindly accepted; `ocaml/testsuite/tests/mixed-blocks/constructor_args.ml` is an exception where I deleted some extensible variant functionality tests. But this testing work is trivial to just recreate (or dig up from git history) in some future world where we allow mixed extensible constructors.

### CODE REVIEW

**Review by goldfirere — APPROVED** [review]

**Comment by ncik-roberts** [comment]
Thanks for approving @goldfirere. I'll let @lthls weigh in before we click "merge".

**Comment by lthls** [comment]
#2533 rebased on this PR passes CI, so it's fine for me.

**Pull request merged**
